services/user/Branding/plugin/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#[allow(warnings)]
+mod bindings;
+
+use bindings::exports::branding::plugin::api::Guest as Api;
+use bindings::exports::branding::plugin::queries::Guest as Queries;
+use bindings::host::common::server as CommonServer;
+use bindings::host::common::types::Error;
+
+use psibase::fracpack::Pack;
+
+mod errors;
+use errors::ErrorType;
+use psibase::Hex;
+
+struct BrandingPlugin;
+
+impl Api for BrandingPlugin {
+    fn set_network_name(name: String) {
+        let packed_network_name_args = branding::action_structs::set_network_name { name }.packed();
+        CommonServer::add_action_to_transaction("set_network_name", &packed_network_name_args)
+            .unwrap();
+    }
+    fn set_logo(logo: Vec<u8>) {
+        let packed_logo_args = branding::action_structs::storeSys {
+            path: String::from("/network_logo.svg"),
+            contentType: String::from("image/svg+xml"),
```

Comment:
- The ui should certainly check, and it's easy to do there. I'll add that.
- By the time we're calling the plugin, the image is already a Uint8Array, so the only "verification" we could do would be for the user to send a separate param with the file name OR go through the cost of decoding the ByteArray assuming it's an svg and look for a successful decoding, all of which I think is minimally helpful.
- The service would be where to do a final check, but again, at that point, we'd have to verify that the byte stream has meta data in it that says it's an SVG. Otherwise, we just store it, assuming it's right, and let the user bear the consequences if it's not.

As for determining SVG format directly from the bytestream, many SVGs start the stream with "<svg", but apparently, they can also be encoded in XML, which is far harder to interpret, leading us to decode the whole file to be sure. and a hacker could just set those initial bytes if the concern was security so...

Potential Concerns we could Address:
- Security: I don't think there's much we can do here other than don't do anything with the bytes but store and serve them.
- Spam: set a max upload size? i think tx/storage costs handle this.
- other?

Thoughts?

Reply:
> so the only "verification" we could do would be for the user to send a separate param with the file name

The file name is unreliable

> OR go through the cost of decoding the ByteArray assuming it's an svg and look for a successful decoding, all of which I think is minimally helpful.

That's what I was thinking. Parsing XML is way too heavyweight to do in the service, so it needs to be done in either the plugin or the ui (I don't really have a strong opinion about which). My main concern is someone not realizing it has to be SVG, uploading a different image format, and getting very confused.

> many SVGs start the stream with "<svg", but apparently, they can also be encoded in XML

It's always XML. `<svg` *is* the beginning of an XML tag.

Reply:
So, the UI *says* to upload an SVG. I could certainly check that what they selected was SVG and warn them, *but* important note: this is an admin app, so I haven't been aiming at putting this kind of polish on something that's an admin tool.
I'm mainly concerned with bigger issues like, e.g., security, which in this case, if we can only enforce effectively in the plugin, that leaves the service vulnerable, so I suggest we do nothing, given we can't (and I don't think need to) handle this thoroughly and completely.

Reply:
Doing validation should ideally done in the plugin, unless it's for some reason significantly simpler to do on the front-end.
Maybe a crate like [this](https://docs.rs/nsvg/0.5.1/nsvg/struct.SvgImage.html#method.parse_str) could work?

Reply:
Ok, I've added an `accept` attrib on the file picker as well as a check once selected.

Reply:
Given this is an operator app, I'm not going to spend the time now adding a lib to detect SVG, especially given the first request we get from users is likely to support some format other than SVG. lol